### PR TITLE
[Feature] Support gradio demo for DragGAN

### PR DESCRIPTION
**Pull request by LeoXing1996**
Support Gradio Demo for this great work!

image

Supported by OpenMMLab [MMagic Team](https://github.com/open-mmlab/mmagic).

### CODE REVIEW

**Comment by hebochang** [comment]
good!

**Comment by radames** [comment]
hi @LeoXing1996 , could you also add a `requirements.txt` to make running gradio app easily? Also do you mind if I publish the demo on https://huggingface.co/? or would you like to do that? thanks

File: visualizer_drag_gradio.py [hunk 1]
```diff
@@ -0,0 +1,866 @@
...
+import numpy as np
+import torch
+from PIL import Image
+
+import dnnlib
+from gradio_utils import (ImageMask, draw_mask_on_image, draw_points_on_image,
+                          get_latest_points_pair, get_valid_mask,
+                          on_change_single_global_state)
+from viz.renderer import Renderer, add_watermark_np
+
```
Comment (radames) [inline visualizer_drag_gradio.py:16]: hi @XingangPan if you'd like I've upload the models to a repository on Hugging Face

```python 
# download models from HuggingFace hub
from huggingface_hub import snapshot_download
model_dir = Path('./checkpoints')
snapshot_download('radames/DragGan', repo_type='model', local_dir=model_dir)
```

Comment (radames) [reply visualizer_drag_gradio.py:16]: I'll be happy to move the repo to @XingangPan

**Comment by radames** [comment]
and for the pip requirements, it's only
```
torch
torchvision
Ninja
gradio
huggingface_hub
hf_transfer
```

**Comment by radames** [comment]
@LeoXing1996 your demos is running here, with a few changes, https://huggingface.co/spaces/radames/DragGan

**Comment by LeoXing1996** [comment]
> @LeoXing1996 your demos is running here, with a few changes, https://huggingface.co/spaces/radames/DragGan

Maybe it would be better to keep the copyright of OpenMMLab. 🤗

**Comment by radames** [comment]
Hi @LeoXing1996 I can add more context on the top of the project?

**Comment by LeoXing1996** [comment]
> Hi @LeoXing1996 I can add more context on the top of the project?

Of course you can, based on preserving the Copyright~ 🤗

**Comment by radames** [comment]
Hi @LeoXing1996 adding back the footer and © for on the top! thanks

**Pull request merged**